crates/binstalk/src/ops/resolve.rs

```diff
@@ -91,45 +92,72 @@ async fn resolve_inner(
         .collect::<Result<Vec<_>, _>>()?;
     let resolvers = &opts.resolvers;
 
-    let mut handles: Vec<(Arc<dyn Fetcher>, _)> =
-        Vec::with_capacity(desired_targets.len() * resolvers.len());
-
-    let data = Arc::new(Data::new(
-        package_info.name.clone(),
-        package_info.version_str.clone(),
-        package_info.repo.clone(),
-    ));
+    let binary_name = match package_info.binaries.as_slice() {
```

Comment:
Could we move this to line 150, to reduce the diff and make the code more readable please?

Other than that, this PR looks good to me!

Reply:
Oh sorry I forgot that it is used to reserve the capacity in the vec